src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -1781,6 +1784,62 @@ public AuditableRestAction<Emote> createEmote(@Nonnull String name, @Nonnull Ico
         });
     }
 
+    @Nonnull
+    @Override
+    public AuditableRestAction<GuildSticker> createSticker(@Nonnull String name, @Nonnull String description, @Nonnull FileUpload file, @Nonnull String tag, @Nonnull String... tags)
+    {
+        checkPermission(Permission.MANAGE_EMOTES_AND_STICKERS);
+        Checks.inRange(name, 2, 30, "Name");
+        Checks.notNull(file, "File");
+        Checks.notEmpty(tag, "Tags");
+        Checks.notNull(description, "Description");
+        if (!description.isEmpty())
+            Checks.inRange(description, 2, 100, "Description");
+        for (String t : tags)
+            Checks.notEmpty(t, "Tags");
+
+        String csv = tag;
+        if (tags.length > 0)
+            csv += "," + String.join(",", tags);
+        Checks.notLonger(csv, 200, "Tags");
+
+
+        // Add sticker meta data as form parts (because payload_json is broken)
+        MultipartBody.Builder builder = new MultipartBody.Builder().setType(MultipartBody.FORM);
+        builder.addFormDataPart("name", name);
+        builder.addFormDataPart("description", description);
+        builder.addFormDataPart("tags", csv);
+
+        // Extract file extension and map to media type
+        int index = file.getName().lastIndexOf('.');
+        if (index < 0)
+            throw new IllegalArgumentException("Missing file extension for image. Must be PNG or JSON.");
```

Comment:
So if someone uses the `byte[]` or `InputStream` ways of building `FileUpload` it is expected that they'll need to provide a `name` with an extension in it?

I guess that makes sense as that's what discord would require to render a preview of the file in the client.

Reply:
The extension is only required because the sticker endpoint expects a valid media type, and we don't want to try and parse it from the data.

Reply:
Maybe we should change the error to be something more descriptive of how to fix it then:
`FileUpload.getName() is missing file extension for image. Must be PNG or JSON.` or something similar?